### PR TITLE
PyTorch AWQ Weight-only 3x API Implementation 

### PR DESCRIPTION
**Pull request by Kaihui-intel**
## Type of Change

feature

## Description

detail description

## Expected Behavior & Potential Risk

the expected behavior that triggered by this PR 

## How has this PR been tested?

how to reproduce the test (including hardware information)

## Dependency Change?

any library dependency introduced or removed

### CODE REVIEW

File: neural_compressor/torch/quantization/config.py [hunk 1]
```diff
@@ -66,6 +67,7 @@ class RTNConfig(BaseConfig):
         "dtype",
         "bits",
         "group_size",
+        # "group_dim", TODO: RuntimeError: group_dim=0
```
Comment (yiliu30) [inline neural_compressor/torch/quantization/config.py:70]: Please remove the comment once the https://github.com/intel/neural-compressor/pull/1595  is merged.

File: neural_compressor/torch/algorithms/weight_only/utility.py [hunk 2]
```diff
@@ -441,3 +441,637 @@ def quant_weight_w_scale(weight, scale, zp, group_size=-1, dtype="int"):
...
+        cnt = 0
+        for idx, input in enumerate(dataloader):
+            output = forward_wrapper(model, input, device)
+            cnt += 1
+            if iters != -1 and cnt >= iters:
+                break
+
+
+# copy from neural_compressor/adaptor/torch_utils/smooth_quant.py
+##TODO potential bug, data typeR
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/utility.py:470]:
```suggestion
# TODO: potential bug, data type
```
Please use `TODO:` as a mark for to-do comments.

File: neural_compressor/torch/algorithms/weight_only/utility.py [hunk 3]
```diff
@@ -441,3 +441,637 @@ def quant_weight_w_scale(weight, scale, zp, group_size=-1, dtype="int"):
             int_weight_tmp.add_(zp[:, -1].unsqueeze(1))
         int_weight[:, leng * group_size :].copy_(int_weight_tmp.round_())
     return int_weight
+
+
+# -------------- AWQ ---------------------------
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/utility.py:446]: If you have any exposed functions or classes, please add them to `__all__` list.

File: neural_compressor/torch/algorithms/weight_only/awq.py [hunk 4]
```diff
@@ -0,0 +1,627 @@
+# Copyright (c) 2023 Intel Corporation
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/awq.py:1]: Please use `2024` for newly added files. Please check other files as well.

File: neural_compressor/torch/algorithms/weight_only/utility.py [hunk 5]
```diff
@@ -441,3 +441,637 @@ def quant_weight_w_scale(weight, scale, zp, group_size=-1, dtype="int"):
...
+        model (torch.nn.Module): input model
+        dataloader: dataloader. Defaults to None.
+        n_samples (int, optional): n_samples. Defaults to 128.
+        calib_func: calib_func. Defaults to None.
+    """
+    # calibration with dataloader or calib_func
+    if calib_func is not None:
+        calib_func(model)
+    else:
+        import math
```
Comment (yiliu30) [inline neural_compressor/torch/algorithms/weight_only/utility.py:938]: Please place the import at the top of the file.

**Review by yiliu30 — APPROVED** [review]
Left some comments; others are LGTM.

File: neural_compressor/torch/quantization/config.py (hunk 1 above)
Comment (Kaihui-intel) [reply neural_compressor/torch/quantization/config.py:70]: got it.

Comment (Kaihui-intel) [reply neural_compressor/torch/quantization/config.py]: remove in https://github.com/intel/neural-compressor/pull/1561/commits/f7b3981744926704d04bf14fbf3134b7d66b0fea

File: neural_compressor/torch/algorithms/weight_only/awq.py (hunk 4 above)
Comment (Kaihui-intel) [reply neural_compressor/torch/algorithms/weight_only/awq.py]: fixed https://github.com/intel/neural-compressor/pull/1561/commits/79021b4e2f667bbe77e353dfe195c5d29a9d668a

**Review by xin3he — APPROVED** [review]

**Pull request merged**